app/assets/javascripts/services/project_roles_svc.js

```diff
@@ -1,6 +1,26 @@
-samson.service('projectRolesService', function($http, $q) {
+samson.service('projectRolesService', function($http, $q, CacheFactory) {
+  /*
+    Creates a cache for project roles to prevent access to the backend each time we need to load the project roles,
+    and corresponding delays to show the data in the browser. Only first request will hit the backend.
+    Data stored in the browser session storage (when available). If not, it will be stored in memory.
+    Cache will expires after 5 minutes (this is a catalog, so, not expect to change that often).
+   */
+  if (!CacheFactory.get('project_roles_cache')) {
+    CacheFactory.createCache('project_roles_cache', {
+      deleteOnExpire: 'passive', //expired items deleted on access
+      maxAge: 5 * 60 * 1000, //expires after 10 minutes
```

Comment:
Damn, I meant to delete this comments after adding the longer description.

Good catch.